### PR TITLE
Create first draft of the use cases document

### PR DESCRIPTION
**Pull request by danielkhan**
This is the first shot at a document that collects use-cases for response headers.

### CODE REVIEW

**Review by mtwo — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

File: response-headers-usecases.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Use Cases for response Headers
+
+The [Response Header Specification](https://github.com/w3c/trace-context/blob/master/spec/21-http_response_header_format.md) defines the format for response headers.
+
+As response headers solve (largely) different use cases, this doc aims to collect them and also address possible implications, like privacy issues.
+
+This document is a loose, unordered and ongoing collection.
+
+## Usecase 1: Report the proposed parent ID back to the caller
```
Comment (jcchavezs) [inline response-headers-usecases.md:9]: While it is good you describe use cases for such things I believe those should be self described. For example, why some systems can't create a trace ID and span ID, why the caller would need a parent ID? Does the parent ID is enough info without the trace ID?

Use case as it is feels like describing the what and not the why.

**Comment by codefromthecrypt** [comment]
https://github.com/w3c/trace-context/issues/406 was intending to remove exactly this

**Review by dyladan — APPROVED** [review]

File: response-headers-usecases.md (hunk 1 above)
Comment (codefromthecrypt) [reply response-headers-usecases.md:9]: This topic has been well discussed and is niche at best. It doesn't deserve the "usecase 1" if anything "honorable mention"

Returning trace IDs is far more frequent for putting traces together without this odd notion of "proposed parent ID". Maybe you can put more routine things here by looking at the issues leading up to this or asking users.

For example, this issue from over 4 years ago catalogs some ideas with real people who were interested https://github.com/openzipkin/b3-propagation/issues/4

File: response-headers-usecases.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Use Cases for response Headers
+
+The [Response Header Specification](https://github.com/w3c/trace-context/blob/master/spec/21-http_response_header_format.md) defines the format for response headers.
+
+As response headers solve (largely) different use cases, this doc aims to collect them and also address possible implications, like privacy issues.
+
+This document is a loose, unordered and ongoing collection.
+
+## Usecase 1: Report the proposed parent ID back to the caller 
+
```
Comment (Kanwaldeep) [inline response-headers-usecases.md:10]: Here is another use case - Currently some of the Azure Control Plane APIs return Correlation Id as part of the response header which can then be used by the customer on reporting an issue or contacting us. We would like to have teams transition to using the standard W3C trace content and returning this as standard response header. Having the trace context in the form of the traceresponse header would make this easy for customers and enable us to pinpoint issues.

**Pull request closed without merging**